src/frontend/src/apis/projects.api.ts

```diff
@@ -89,3 +89,12 @@ export const getAllLinkTypes = () => {
     transformResponse: (data) => JSON.parse(data).map(linkTypeTransformer)
   });
 };
+
+/**
+ * Creates a linkType in the database
+ * @param linkType the data for the linkType
+ * @returns the new linkType
+ */
+export const createLinkType = async (linkTypeData: { name: string }) => {
```

Comment:
I would declare this type you made as LinkTypeCreatePayload and use it here and in the hook

Reply:
also there should be a lot more fields in the create payload, look at the create link type endpoint